homework15/js/main.js

```diff
@@ -0,0 +1,156 @@
+let locationKey;
+
+const optionLocation = document.querySelector('.weather__location');
+const buttonGetWeather = document.querySelector('.weather__btn');
+
+function getLocationKey(){
+    switch (optionLocation.value) {
+        case "Lviv":
+            locationKey = locationCode[0];
+            break;
+        case "Kyiv" :
+            locationKey = locationCode[1];
+            break;
+        case "Kharkiv":
+            locationKey = locationCode[2];
+            break;
+        case "Chernihiv":
+            locationKey = locationCode[3];
+            break;
+        case "Odesa":
+            locationKey = locationCode[4];
+            break;
+        case "Mariupol":
+            locationKey = locationCode[5];
+            break;
+        case "Poltava":
+            locationKey = locationCode[6];
+            break;
+        case "Zhytomyr":
+            locationKey = locationCode[7];
+            break;
+        case "Cherkasy":
+            locationKey = locationCode[8];
+            break;
+        case "London":
+            locationKey = locationCode[9];
+            break;
+        case "Warsaw":
+            locationKey = locationCode[10];
+            break;
+        case "New York":
+            locationKey = locationCode[11];
+            break;
+    }
+    return locationKey;
+}
+
+//Функція переведення температури зі шкали Фаренгейта до Цельсія
+function temperatureConversion(temperature){
+    return Math.floor((temperature -  32) / 1.8);
+}
+
+const xhr = new XMLHttpRequest();
+
+function loadData() {
+    xhr.open('GET', `http://dataservice.accuweather.com/forecasts/v1/daily/5day/${getLocationKey()}?apikey=${API_KEY}`);
+    xhr.send();
+    xhr.onload = function () {
+        if (this.readyState === 4 && this.status === 200) {
+            const result = JSON.parse(this.response);
+
+            for(const weather of result["DailyForecasts"]){
+
+                //Додаємо дату
+                const dateElement = document.createElement('p');
+                dateElement.className = 'weather__date';
+
+                //Обираємо день
+                let day = new Date(weather.Date).getDay();
+                switch (day){
+                    case 0:
+                        day = 'Sunday';
+                        break;
+                    case 1:
+                        day = 'Monday';
+                        break;
+                    case 2:
+                        day = 'Tuesday';
+                        break;
+                    case 3:
+                        day = 'Wednesday';
+                        break;
+                    case 4:
+                        day = 'Thursday';
+                        break;
+                    case 5:
+                        day = 'Friday';
+                        break;
+                    case 6:
+                        day = 'Saturday';
+                        break
+                }
+
+                //Додаємо та витягуємо необхідну інформацію з дати
```

Comment:
Nice to clear data for the previous request before showing the next one


https://user-images.githubusercontent.com/84010501/201979171-c43060ea-6a38-4c27-beac-f93a6272fc04.mov